sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/util/LoggingUtil.java

```diff
@@ -84,15 +87,15 @@ private LoggingUtil() {
 
     public static CredentialUnavailableException logCredentialUnavailableException(ClientLogger logger,
                                        IdentityClientOptions options, CredentialUnavailableException exception) {
-        logger.log(options.getIdentityLogOptions().getRuntimeExceptionLogLevel(),
-            () -> exception.getMessage(), exception);
+        logger.log(options.getIdentityLogOptions().getRuntimeExceptionLogLevel(), exception::getMessage, exception);
         return exception;
     }
 
     public static void logError(ClientLogger logger, IdentityClientOptions options, String message) {
-        logger.log(options.getIdentityLogOptions().getRuntimeExceptionLogLevel(),
-            () -> message);
-        return;
+        logger.log(options.getIdentityLogOptions().getRuntimeExceptionLogLevel(), () -> message);
     }
 
+    public static void logError(ClientLogger logger, IdentityClientOptions options, Supplier<String> messageSupplier) {
```

Comment:
Added this as there were many places that were eagerly calling String.format when it could be possible that logging wouldn't happen. This defers String.format until it is known that logging will occur.